### PR TITLE
Add list expressions

### PR DESCRIPTION
**Pull request by etiennebacher**

### CODE REVIEW

File: src/rust/src/conversion/mod.rs [hunk 1]
```diff
@@ -356,10 +357,46 @@ impl TryFrom<&str> for Wrap<WindowMapping> {
...
+        let parsed = match operation {
+            "union" => SetOperation::Union,
+            "intersection" => SetOperation::Intersection,
+            "difference" => SetOperation::Difference,
+            "symmetric_difference" => SetOperation::SymmetricDifference,
+            v => {
+                return Err(format!(
+                    "`operation` must be one of 'union', 'intersection', 'difference', 'symmetric_difference'. Got '{v}'",
+                ))
+            }
```
Comment (etiennebacher) [inline src/rust/src/conversion/mod.rs:381]: This should be done on the R side with `arg_match()` instead, so it could maybe be `unreachable!()` here?

Comment (eitsupi) [reply src/rust/src/conversion/mod.rs:381]: I don't think we need to use the unreachable macro, but I don't think we need to pretty the comment (perhaps something like `unexpected input` or something is good enough.)

File: src/rust/src/datatypes.rs
Comment (eitsupi) [inline src/rust/src/datatypes.rs:1]: Are these changes needed?

**Review by eitsupi — CHANGES_REQUESTED** [review]
Please rebase.

**Comment by etiennebacher** [comment]
Aaah git is killing me

**Comment by etiennebacher** [comment]
I don't have the "Merge main" or "Rebase" buttons on this repo, can you give me the rights for those?

**Comment by eitsupi** [comment]
Hmmm, I don't think rebasing on GitHub is easier to do than rebasing locally, so it seems difficult to perform that operation if you are not familiar with local Git operations.
Why not abandon this branch and just copy the necessary files to the new branch?

**Comment by etiennebacher** [comment]
superseded by #12
